sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -64,6 +65,19 @@ class SparkSession private(
    |  Session-related state  |
    * ----------------------- */
 
+  {
+    val defaultWarehousePath =
+      SQLConf.WAREHOUSE_PATH
+        .defaultValueString
+        .replace("${system:user.dir}", System.getProperty("user.dir"))
+    val warehousePath = sparkContext.conf.get(
+      SQLConf.WAREHOUSE_PATH.key,
+      defaultWarehousePath)
+    sparkContext.conf.set(SQLConf.WAREHOUSE_PATH.key, warehousePath)
+    sparkContext.conf.set("hive.metastore.warehouse.dir", warehousePath)
```

Comment:
Both problems are fixed in the PR: https://github.com/apache/spark/pull/13128. Let me know if my fix is appropriate. Thanks!